uberfire-message-console/uberfire-message-console-client/src/main/java/org/guvnor/messageconsole/client/console/MessageConsoleServiceRow.java

```diff
@@ -16,25 +16,43 @@
 
 package org.guvnor.messageconsole.client.console;
 
+import java.util.Comparator;
+
 import org.guvnor.common.services.shared.message.Level;
 import org.guvnor.messageconsole.events.SystemMessage;
 import org.uberfire.backend.vfs.Path;
 import org.uberfire.paging.AbstractPageRow;
 
 public class MessageConsoleServiceRow extends AbstractPageRow {
 
-    String sessionId;
+    private static int COUNTER = 0;
+
+    private final long sequence;
+
+    private String sessionId;
+
+    private String userId;
 
-    String userId;
+    private SystemMessage message;
 
-    SystemMessage message;
+    static final Comparator<MessageConsoleServiceRow> DESC_ORDER = (o1, o2) -> {
+        final long sequence1 = o1.sequence;
+        final long sequence2 = o2.sequence;
+        if (sequence1 < sequence2) {
+            return 1;
+        } else if (sequence1 > sequence2) {
+            return -1;
+        }
+        return 0;
+    };
```

Comment:
I believe this anonymous comparator can be written as:
```java
static final Comparator<MessageConsoleServiceRow> NATURAL_ORDER = Comparator.comparingLong(row -> row.sequence);
    
static final Comparator<MessageConsoleServiceRow> DESC_ORDER = NATURAL_ORDER.reversed();
```

Reply:
@jomarko I'll give it a try (IDK if GWT supports all of what you propose).

Reply:
Thanks @manstis 

P.S. Dear GWT, please start support all usable stuff :D